istioctl/cmd/testdata/webhook/example-mutating-webhook-config.yaml

```diff
@@ -26,3 +26,4 @@ webhooks:
         resources:
           - pods
     sideEffects: Unknown
+    admissionReviewVersions: ["v1", "v1beta1"]
```

Comment:
My understanding was that it would support both `v1` and `v1beta1` objects if added. I am guessing this ensures that backcompat when istio users are transitioning to the new version?

Reply:
I believe @elfinhe said that this was required

Reply:
I guess it's fair to expect that 1.7 will not have any `v1beta1` webhooks.

Honestly, I find this apiversion stuff really confusing right now.

Reply:
```
	// If a persisted webhook configuration specifies allowed versions and does not
	// include any versions known to the API Server, calls to the webhook will fail
	// and be subject to the failure policy.
	// Default to `['v1beta1']`.
```

Even if `v1` is not supported, it will always fall back to `v1beta1` since that is recognized by the APIServer. This was my understanding from the API doc

Reply:
This is about versions know to the API server though. k8s 1.16-1.22 will support both v1beta1 AND v1. Istio 1.6 will support only v1beta1, and Istio 1.7+ *should* support v1beta1 and v1 ideally?

Reply:
> API servers send the first AdmissionReview version in the admissionReviewVersions list they support. If none of the versions in the list are supported by the API server, the configuration will not be allowed to be created. 

Here is the k8s doc: https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#webhook-request-and-response.

Reply:
> This is about versions know to the API server though. k8s 1.16-1.22 will support both v1beta1 AND v1. Istio 1.6 will support only v1beta1, and Istio 1.7+ _should_ support v1beta1 and v1 ideally?

yes, Istio 1.7+ need v1beta1 and v1 to be able to upgrade to the first version (in Istio 1.8?) using v1.

Reply:
So don't we need to upgrade the client-go code to using V1 APIs before making this change?